### ISSUE TITLE
 TypeError: Cannot read property '1' of null at exports.handler (/var/task/index.js:16:20)

### ISSUE THREAD

**Issue by concalahan** [sagidM/s3-resizer#9]
Hello,

I cannot create images.

<img width="1217" alt="Screen Shot 2019-11-09 at 3 44 37 PM" src="https://user-images.githubusercontent.com/24490038/68525798-fc411700-0307-11ea-9cb0-4ecf6a5877ae.png">

**Comment by sagidM** [comment]
@concalahan I can see your image, however, there is no redirect to your lambda.
1. Does this work for other images?
2. Have you set redirects to your lambda? YOUR_BUCKET -> Properties ->  Static website hosting as it shown in the image
![image](https://user-images.githubusercontent.com/13490595/69100534-aeda4d80-0a5d-11ea-8ff4-51cf31398d33.png)

**Comment by fabianmossberg** [comment]
I have the same problem:

It works with files in the root:

Original: http://photos.matchatheapp.com.s3-website-eu-west-1.amazonaws.com/horse.jpg
Resize: http://photos.matchatheapp.com.s3-website-eu-west-1.amazonaws.com/100x100/horse.jpg


It does not work with files in directories:

Original: https://s3-eu-west-1.amazonaws.com/photos.matchatheapp.com/profiles/XrPLrK/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg
Resize: https://s3-eu-west-1.amazonaws.com/photos.matchatheapp.com/100x100/profiles/XrPLrK/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg

**Comment by fabianmossberg** [comment]
Realized that te resize might need to be just before the image:

https://s3-eu-west-1.amazonaws.com/photos.matchatheapp.com/profiles/XrPLrK/100x100/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg

Doesn't work with this either.

**Comment by sagidM** [comment]
@fabianmossberg Could you please confirm that the images have been resized, but you get **AccessDenied** when you request them?
I.e. you have this image `profiles/XrPLrK/100x100/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg`

**Comment by fabianmossberg** [comment]
@sagidM : no, the folder 100x100 does not exist in the `profiles/XrPLrK/` directory.

**Comment by sagidM** [comment]
@fabianmossberg First, please check out if `https://photos.matchatheapp.com.s3-eu-west-1.amazonaws.com` is the link of the Endpoint of your Static Website Hosting https://github.com/sagidM/s3-resizer/issues/9#issuecomment-555246339, not the link of your bucket.
As Static Website Hosting does not provide HTTPS https://github.com/sagidM/s3-resizer/issues/7.
(I am a bit confused about mixing the `http` and `https` protocols)

Also, there is an item `Test your lambda` in the readme.
What can you see when you test your lambda with the parameters?
```json
{
  "queryStringParameters": {"path": "profiles/XrPLrK/100x100/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg"}
}
```
If it fails, the log of the lambda or the screenshot will definitely help.

**Comment by fabianmossberg** [comment]
@sagidM : this is strange. Today it works. Even the links I Posted as "not working" yesterday actually produces the expected result today. I'm not sure what I did wrong.

But I do see some errors I had done.

When I filled the "path" in the test you had in the example, I filled the whole URL, not just the path.

I also realized the difference between the static hosting URL
```photos.matchatheapp.com.s3-website-eu-west-1.amazonaws.com```
and the Bucket URL.
```photos.matchatheapp.com.s3-eu-west-1.amazonaws.com````

They look so alike that I completely missed that they were different domains.

**Comment by sagidM** [comment]
@fabianmossberg Yeah, I believe you, I have been keeping the tab from yesterday with an error.
![image](https://user-images.githubusercontent.com/13490595/75609096-ddfd6200-5b16-11ea-8749-7356d3d64129.png)

And I have already seen this sort of error when the accessibility to a file was wrong. For example, if the bucket policy has not been provided or a role's policy for a lambda was wrong
(`"Resource": "arn:aws:s3:::__BUCKET_NAME__/*"` in readme).

But if it started working without any change, that is really strange.

So, let's keep this issue open for a while and if you encounter the same problem, please do tests and provide the output.

**Comment by sagidM** [comment]
Oops, it is still not working
http://photos.matchatheapp.com.s3-eu-west-1.amazonaws.com/profiles/XrPLrK/50x50/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg
does not work. Please, tinker with different params, for example
`{
  "queryStringParameters": {"path": "profiles/XrPLrK/99x99/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg"}
}`
`{
  "queryStringParameters": {"path": "profiles/XrPLrK/95x95/51ED545C-AB28-4988-9D6B-598A0DD6A02F.jpeg"}
}`

**Comment by 5URE5H** [comment]
@sagidM how to fix this?

**Comment by sagidM** [comment]
@5URE5H how much does it take to the point when it stops working?

**Comment by 5URE5H** [comment]
> @5URE5H how much does it take to the point when it stops working?

For nested items, URL gives error when I try to access it right off the bat, once I test in lambda (for nested cases) then URL is working without any issues from then on. If it isn't nested inside a folder it works from the start.

**Comment by sagidM** [comment]
@5URE5H would you like to make a call some time?
btw, I use telegram: https://t.me/sagidM

**Comment by philyboysmith** [comment]
Hey @sagidM idd you ever get to the bottom of this? I'm trying to get this working with folders, but no luck so far.

**Comment by sagidM** [comment]
@philyboysmith unfortunately, I could not reproduce the bug. The images are stored in to directories

**Comment by paoyangt** [comment]
Hi, I'm facing the same issue when trying to resize images that are nested within the bucket. I've tested with the lambda url, for example: 
https://lambda-endpoint.com/resize?path=150x150/image.png <== works without issues.
https://lambda-endpoint.com/resize?path=150x150/nested/image.png <== Exception: Access Denied 

I've checked my bucket and confirmed that the nested image was not resized at all while the one on root path was successfully resized. 

Are there any updates for this issue?

**Comment by sagidM** [comment]
@paoyangt hi, can you share your policy config?

**Comment by paoyangt** [comment]
Here it is

`{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents"
            ],
            "Resource": "arn:aws:logs:*:*:*"
        },
        {
            "Effect": "Allow",
            "Action": "s3:PutObject",
            "Resource": [
                "arn:aws:s3:::x3-resized-thumbnails/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": "s3:GetObject",
            "Resource": [
                "arn:aws:s3:::x3-resized-thumbnails/*"
            ]
        }
    ]
}`

**Comment by paoyangt** [comment]
Here it is @sagidM 

`{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents"
            ],
            "Resource": "arn:aws:logs:*:*:*"
        },
        {
            "Effect": "Allow",
            "Action": "s3:PutObject",
            "Resource": [
                "arn:aws:s3:::resized-thumbnails/*"
            ]
        }
    ]
}`

**Comment by condrowiyono** [comment]
mine working fine.

```
https://lambda-endpoint.com/resize?path=150x150/image.png <== works without issues.
https://lambda-endpoint.com/resize?path=nested/150x150/image.png <= beware that directory / folder must come first
https://lambda-endpoint.com/resize?path=nested/nested/150x150/image.png <= beware that directory / folder must come first

```

So, basically the formula would be `"anything-dir/you/have/{150x150}/{filename.ext}`
